pkg/alertdog/prometheus.go

```diff
@@ -0,0 +1,70 @@
+package alertdog
+
+import (
+	"time"
+
+	"github.com/errm/alertdog/pkg/alertmanager"
+	"github.com/prometheus/alertmanager/template"
+)
+
+type AlertAction int
+
+const (
+	ActionNone AlertAction = iota
+	ActionAlert
+	ActionResolve
+)
+
+type Prometheus struct {
+	MatchLabels  map[string]string `yaml:"match_labels"`
+	Expiry       time.Duration
+	PagerDutyKey string `yaml:"pager_duty_key"`
+	Alert        alertmanager.Alert
+	checkedIn    time.Time
+	count        uint
+}
+
+func (p *Prometheus) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	defaultExpiry, _ := time.ParseDuration("4m")
+	p.Expiry = defaultExpiry
+	type plain Prometheus
+	return unmarshal((*plain)(p))
+}
+
+func (p *Prometheus) CheckIn(alert template.Alert) AlertAction {
+	if p.match(alert.Labels) {
+		if alert.Status == "firing" {
+			p.checkedIn = time.Now()
+			p.count += 1
+			// Debounce during state change, wait for 2 alerts before resolving
+			if p.count == 2 {
```

Comment:
❓ What is the reasoning behind `2` here? Is there any specific reason we've pick up this number here?

Reply:
Because it is usual to run allertmanger in a ha pair.

If they are slightly out of sync you can get a flap when wen the watchdog goes from "firing" to "resolved" (i.e. prometheus wen't away)

for example I saw this scenario several times in my testing:
```
am1: firing
am2 firing
am1: resolved -> we trigger the alert
am2: firing (still) -> we resolve the alert
am2: resolved -> we trigger the alert (again)
```

I will add a test case for this!

Reply:
I had the same question while I was reviewing. So, if we specify 3 Alertmanager with `--cluster.peer=` options, does this become `3`?